src/bench_generic/input_generators.h

```diff
@@ -33,62 +33,139 @@
 #include "algo/nth_permutation.h"
 #include "algo/shuffle_biased.h"
 #include "algo/type_functions.h"
+#include "algo/uint_tuple.h"
 #include "bench_generic/fake_url.h"
 #include "bench_generic/noinline_int.h"
 
 namespace bench {
 
+using uint32 = std::uint32_t;
+using uint64 = std::uint64_t;
+
 using std_int64_t = std::int64_t;
 
 using fake_url_pair = std::pair<fake_url, fake_url>;
 
+using uint_std_pair32 = std::pair<std::uint32_t, std::uint32_t>;
+using uint_std_pair64 = std::pair<std::uint64_t, std::uint64_t>;
+
+using uint_tuple_pair32 = algo::uint_tuple<32, 32>;
+using uint_tuple_pair64 = algo::uint_tuple<64, 64>;
+
+struct use_pair {
+  template <size_t size>
+  using type = std::pair<algo::uint_t<size>, algo::uint_t<size>>;
+
+  template <typename T, typename U>
+  constexpr auto operator()(T x, U y) const {
+    return std::pair{x, y};
+  }
+};
+
+struct use_uint_tuple {
+  template <size_t size>
+  using type = algo::uint_tuple<size, size>;
+
+  template <typename T, typename U>
+  constexpr auto operator()(T x, U y) const {
+    algo::uint_tuple<algo::uint_bit_size<T>(), algo::uint_bit_size<U>()> res;
+    algo::set_at<0>(res, x);
+    algo::set_at<1>(res, y);
```

Comment:
Why are you manually calling `set_at` instead of using template variadic constructor?

Reply:
It wasn't available when i wrote this - will change.

Reply:
Done